src/api/deposit.js

```diff
@@ -1,21 +1,19 @@
 const { post } = require('request-promise')
 const DVFError = require('../lib/dvf/DVFError')
 const validateAssertions = require('../lib/validators/validateAssertions')
-const { Joi } = require('dvf-utils')
-
-const schema = Joi.object({
-  amount: Joi.amount().required(), // number or number string
-})
+const prepareAmount = require('dvf-utils').prepareAmount
+const maxQuantizedDecimalPlaces = require('../lib/dvf/token/maxQuantizedDecimalPlaces')
 
 module.exports = async (dvf, token, amount, starkPrivateKey) => {
   validateAssertions(dvf, { amount, token, starkPrivateKey })
-  const { value } = schema.validate({amount})
-  amount = value.amount
-  //TODO: assess and replace all validations with custom Joi
-  const currency = dvf.token.getTokenInfo(token)
 
-  const quantisedAmount = dvf.token.toQuantizedAmount(token, amount)
+  amount = prepareAmount(
+    amount,
+    Math.min(maxQuantizedDecimalPlaces(token), dvf.config.defaultDecimalPlaces)
```

Comment:
I'm not sure if we need to have defaultDecimalPlaces? Shouldnt we just pass maxQuantizedDecimalPlaces? If its more than 8 it will use 8 anyway in dvf-utils

Reply:
Requires less from the user, and one less config variable

Reply:
yes, now this is taken care of by the dvf-utils change 👍